### PR TITLE
gh-121554: remove unnecessary internal functions in compile.c

### LINKED ISSUE #121554: remove  _PyCompile_OpcodeHas* functions
There are a bunch of functions in compile.c that expose data from pycore_opcode_metadata.h as private C API. These are only used in Modules/_opcode.c.

However, Modules/_opcode.c can just include  pycore_opcode_metadata.h and pycore_opcode_utils.h, and access this data directly. 

### Linked PRs
* gh-121555

### PR DESCRIPTION
**Pull request by iritkatriel**
Fixes #121554.

* Issue: gh-121554

### CODE REVIEW

File: Include/internal/pycore_compile.h [hunk 1]
```diff
@@ -76,15 +76,6 @@ int _PyCompile_ConstCacheMergeOne(PyObject *const_cache, PyObject **obj);
 
 
 // Export for '_opcode' extension module
```
Comment (erlend-aasland) [inline Include/internal/pycore_compile.h:78]: Also remove the above comment?

File: Modules/Setup.bootstrap.in [hunk 2]
```diff
@@ -30,6 +30,7 @@ _weakref _weakref.c
 _abc _abc.c
 _functools _functoolsmodule.c
 _locale _localemodule.c
+_opcode _opcode.c
```
Comment (erlend-aasland) [inline Modules/Setup.bootstrap.in:33]: You should remove the `PY_STDLIB_MOD_SIMPLE` def for `_opcode` from `configure.ac` now; it's not needed anymore.

**Review by erlend-aasland — COMMENTED** [review]
Consider splitting this PR in two:

1. the build change
2. the C API change

**Comment by erlend-aasland** [comment]
I wonder if `_PyImport_Inittab` also needs to change now.

File: Include/internal/pycore_compile.h (hunk 1 above)
Comment (iritkatriel) [reply Include/internal/pycore_compile.h:78]: It's true for the two remaining functions about intrinsics.

**Comment by iritkatriel** [comment]
> I wonder if `_PyImport_Inittab` also needs to change now.

It has both:

 ```
   {"_opcode", PyInit__opcode},
    {"_operator", PyInit__operator},
```

so it seems to cover both linkage methods.

**Comment by iritkatriel** [comment]
> Consider splitting this PR in two:
> 
> 1. the build change
> 2. the C API change

If we need to revert the build change then we also need to revert the code change. Might be easier in one PR.

**Comment by erlend-aasland** [comment]
Yeah, but `_opcode` currently comes after the second freeze marker, FWIW.

**Comment by iritkatriel** [comment]
> Yeah, but `_opcode` currently comes after the second freeze marker, FWIW.

What does that mean?

**Comment by erlend-aasland** [comment]
> > Yeah, but `_opcode` currently comes after the second freeze marker, FWIW.
> 
> What does that mean?

Ah, forget it; I was looking at a generated file. Sorry 'bout the noise.

**Comment by iritkatriel** [comment]
@erlend-aasland Thanks for your help. Tests pass now, are you happy to approve this?

File: Modules/_opcode.c [hunk 3]
```diff
@@ -425,15 +418,15 @@ opcode_functions[] =  {
 };
 
 int
-_opcode_exec(PyObject *m) {
+_Py_opcode_exec(PyObject *m) {
```
Comment (erlend-aasland) [inline Modules/_opcode.c]: Why this name change? It looks to me it's only used in L429, right?

Comment (iritkatriel) [reply Modules/_opcode.c]: smelly was complaining about this.

Comment (erlend-aasland) [reply Modules/_opcode.c]: Ah, it's probably because it's not declared as `static`. It should be a file local function.

Comment (iritkatriel) [reply Modules/_opcode.c]: I'll try that.

**Review by erlend-aasland — APPROVED** [review]
Thanks!

**Pull request merged**